tooling/test/validate.go

```diff
@@ -9,75 +10,92 @@ import (
 	"github.com/ipfs/gateway-conformance/tooling/check"
 )
 
+type AnyOfResponse struct {
+	responses []ExpectBuilder
+}
+
+type testCheckOutput struct {
+	testName    string
+	checkOutput check.CheckOutput
+}
+
 func validateResponse(
 	t *testing.T,
 	expected ExpectBuilder,
 	res *http.Response,
-	localReport Reporter,
-) {
+) []testCheckOutput {
```

Comment:
We don't do the reporting here anymore so that it can be controlled by the caller (e.g. for `AnyOf` we don't fail the test unless all the options fail)